mlflow/metrics/metric_definitions.py

```diff
@@ -254,47 +254,47 @@ def _rougeLsum_eval_fn(predictions, targets, metrics):
         )
 
 
-def _mae_eval_fn(predictions, targets, metrics, sample_weight=None):
+def _mae_eval_fn(predictions, targets=None, metrics=None, sample_weight=None):
     if targets is not None and len(targets) != 0:
         from sklearn.metrics import mean_absolute_error
 
         mae = mean_absolute_error(targets, predictions, sample_weight=sample_weight)
         return MetricValue(aggregate_results={"mean_absolute_error": mae})
 
 
-def _mse_eval_fn(predictions, targets, metrics, sample_weight=None):
+def _mse_eval_fn(predictions, targets=None, metrics=None, sample_weight=None):
     if targets is not None and len(targets) != 0:
         from sklearn.metrics import mean_squared_error
 
         mse = mean_squared_error(targets, predictions, sample_weight=sample_weight)
         return MetricValue(aggregate_results={"mean_squared_error": mse})
 
 
-def _rmse_eval_fn(predictions, targets, metrics, sample_weight=None):
+def _rmse_eval_fn(predictions, targets=None, metrics=None, sample_weight=None):
     if targets is not None and len(targets) != 0:
         from sklearn.metrics import mean_squared_error
 
         rmse = mean_squared_error(targets, predictions, squared=False, sample_weight=sample_weight)
         return MetricValue(aggregate_results={"root_mean_squared_error": rmse})
 
 
-def _r2_score_eval_fn(predictions, targets, metrics, sample_weight=None):
+def _r2_score_eval_fn(predictions, targets=None, metrics=None, sample_weight=None):
     if targets is not None and len(targets) != 0:
         from sklearn.metrics import r2_score
 
         r2 = r2_score(targets, predictions, sample_weight=sample_weight)
         return MetricValue(aggregate_results={"r2_score": r2})
 
 
-def _max_error_eval_fn(predictions, targets, metrics):
+def _max_error_eval_fn(predictions, targets=None, metrics=None):
```

Comment:
Why do we need to explicitly set this to None? If users create a custom_metric and not set this to None. Would the error handling work? 

Also in some of the metrics, targets is actually required since we would do a no-op if target is not passed in right? 🤔

Reply:
Yes, we want to do a no-op if targets is None and skip logging that metric (if there is no default it will fail the evaluation saying that 'metric_name' requires parameter 'targets')

Reply:
The reason for this was that _get_args_for_metrics will surface errors for other required columns (such as 'context') but it will not do the same for targets right now - missing 'targets' column (for instance for answer similarity) won't be surfaced until we actually call eval_fn which will raise an ambiguous error since we don't know if targets is "required" or not until then